colony/branch_utils.py

```diff
@@ -172,21 +179,23 @@ def wait_and_then_delete_branch(sb_manager: SandboxesManager, sandbox_id, repo,
     prep_art_status = progress.get("preparing_artifacts").get("status")
     logger.debug("Waiting for sandbox (id={}) to prepare artifacts (before deleting temp branch)...".format(sandbox_id))
 
-    while (datetime.datetime.now() - start_time).seconds < TIMEOUT * 60:
+    BaseCommand.info("Waiting for the Sandbox to start with local changes. This may take some time.")
+    BaseCommand.fyi_info("Canceling or exiting before the process completes may cause the sandbox to fail")
 
-        if status in FINAL_SB_STATUSES or (status == "Launching" and prep_art_status != "Pending"):
-            delete_temp_branch(repo, temp_branch)
-            break
-        else:
-            time.sleep(10)
-            BaseCommand.message(
-                f"Waiting for the Sandbox ({sandbox_id}) to start with local changes..."
-                f"[{int((datetime.datetime.now() - start_time).total_seconds())} sec]"
-            )
-            sandbox = sb_manager.get(sandbox_id)
-            status = getattr(sandbox, "sandbox_status")
-            progress = getattr(sandbox, "launching_progress")
-            prep_art_status = progress.get("preparing_artifacts").get("status")
+    with yaspin(text="Starting", color="yellow") as spinner:
+
+        while (datetime.datetime.now() - start_time).seconds < TIMEOUT * 60:
+
+            if status in FINAL_SB_STATUSES or (status == "Launching" and prep_art_status != "Pending"):
+                delete_temp_branch(repo, temp_branch)
+                break
+            else:
+                time.sleep(10)
+                spinner.text = f"[{int((datetime.datetime.now() - start_time).total_seconds())} sec]"
```

Comment:
I agree with the comment. I think we need to do a refactoring around that (main loop and different flows in the code there) I *don't* want to include it in this story though (since this PR is too big any way)
I created a separate issue for that #36